### PR TITLE
 #187300250:marking Items as available/unavailable

### PR DESCRIPTION
**Pull request by janet-barbie**
#### What does this PR do?
This PR implements the task related to marking items as available or not and A User Should be able to list all Items . It addresses the requirements outlined in Pivotal ID #187300250 , #187300253

#### Description of Task to be completed?
Products should be availed for buyers to purchase
As a seller I should be able to mark a product as available or unavailable manually in order to make it available for the buyers or not.
 As a buyer / admin / guest I should be able to retrieve a list of all available items / a single available item
#### How should this be manually tested?
• use  postman to test the  product status endpoint (PATCH: /api/products/{productId}/status)
• use  postman to test get  single available products endpoint (GET: /api/products/{productId})
• use  postman to test get   available products endpoint (GET: /api/products/)
#### What are the relevant pivotal tracker/Trello stories?
Pivotal ID #187300250
 Pivotal ID #187300253

### CODE REVIEW

**Review by kayigmb — APPROVED** [review]
LGTM!

**Review by kayigmb — APPROVED** [review]

**Review by k3lly003 — APPROVED** [review]

File: src/controllers/product.controllers.ts [hunk 1]
```diff
@@ -73,22 +73,45 @@ export const createProducts = async (req: Request, res: Response) => {
 
 /* this function will help to lists all product items in seller collection */
 
-export const getAllSellerProducts = async (req: Request, res: Response) => {
+export const getAllSellerProducts = async (req: any, res: Response) => {
```
Comment (Eli250) [inline src/controllers/product.controllers.ts:76]: Why did you change the Request to any?

File: src/models/champs-ec-be.code-workspace [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	"folders": [
+		{
+			"path": "../.."
+		},
+		{
+			"path": "../../../../../champs-ec-be"
+		}
+	],
+	"settings": {}
+}
```
Comment (Eli250) [inline src/models/champs-ec-be.code-workspace:11]: what's the use of this file?

**Review by Eli250 — APPROVED** [review]
